src/main/java/de/tum/in/www1/artemis/service/iris/model/IrisGPT3_5Service.java

```diff
@@ -107,16 +108,32 @@ public CompletableFuture<String> getResponse(IrisSession session) {
 
             // TODO: Create DTO for error responses, and handle them
             OpenAIChatResponseDTO response = jsonMapper.readValue(responseBody, OpenAIChatResponseDTO.class);
-            String content = response.getChoices().get(0).getMessage().getContent();
-
-            return CompletableFuture.completedFuture(content);
+            var irisMessage = convertToIrisMessage(response.getChoices().get(0).getMessage());
+            return CompletableFuture.completedFuture(irisMessage);
         }
         catch (JsonProcessingException e) {
             log.error(e.getMessage(), e);
             return CompletableFuture.failedFuture(e);
         }
     }
 
+    /**
+     * Converts the response from the GPT-3.5 API into an {@link IrisMessage} object.
+     *
+     * @param message the response from the GPT-3.5 API
+     * @return the converted {@link IrisMessage}
+     */
+    private IrisMessage convertToIrisMessage(OpenAIChatResponseDTO.Message message) {
```

Comment:
Would love to see this addressed more elegantly in the future, such as with the builder pattern or just a nice factory method. It shouldn't require 6 lines of code to create a new IrisMessage, we can hold ourselves to a higher standard than that.

Reply:
Honestly I don't see how any of those patterns would reduce the amount of code we need here. Overall the total amount of LOCs would actually increase for a (currently) singular use case.
Both patterns would at best put the lines into a single one, which isn't really any better per se. This is mainly due to the fact, that `OpenAIChatResponseDTO.Message` should not be handed around all that much. It is merely a DTO which should be converted to a non DTO object before being passed to other subsystems, which is exactly what is done here.

Reply:
You are right, it would increase the number of lines of code in total. For now it's fine.